gtsfm/view_graph_estimator/view_graph_estimator_base.py

```diff
@@ -0,0 +1,123 @@
+"""Implements a base class for ViewGraph estimation.
+
+Estimating the ViewGraph can be done trivially by adding all the two-view estimates into a ViewGraph data structure.
+The purpose of this class, however, is to define an API for more sophisticated methods for estimating a ViewGraph 
+that include filtering or optimizing the two-view estimates.
+
+Authors: Akshay Krishnan, Ayush Baid
+"""
+import abc
+from typing import Dict, Optional, List, Tuple
+
+import dask
+import numpy as np
+from dask.delayed import Delayed
+from gtsam import Cal3Bundler, PinholeCameraCal3Bundler, Rot3, Unit3
+
+import gtsfm.utils.geometry_comparisons as comp_utils
+from gtsfm.evaluation.metrics import GtsfmMetric, GtsfmMetricsGroup
+from gtsfm.common.keypoints import Keypoints
+from gtsfm.common.view_graph import ViewGraph
+
+METRIC_GROUP = "view_graph"
+
+
+class ViewGraphEstimatorBase(metaclass=abc.ABCMeta):
+    """Base class for ViewGraph estimation.
+
+    A ViewGraphEstimator aggregates two-view estimates into a ViewGraph.
+    It could also improve the two-view estimates using filtering or optimization techniques.
+    """
```

Comment:
Is this module based off of another implementation you've seen. If so, could you add some references here?

Reply:
It’s not based on another implementation Ive seen. The term ViewGraph is borrowed/inspired from other papers. Have you seen similar modules? Happy to add references if you’ve seen any.

Reply:
https://github.com/AIBluefisher/DAGSfM/blob/dev/src/graph/view_graph.h